service/src/test/java/uk/gov/hmcts/reform/fpl/service/OrganisationServiceTest.java

```diff
@@ -87,4 +92,40 @@ private Users prepareUsersForAnOrganisation() {
                 .build()
         ));
     }
+
+    @Test
+    void shouldFindUser() {
+        User user = new User(RandomStringUtils.randomAlphanumeric(10));
+
+        when(organisationApi.findUsersByEmail(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID, USER_EMAIL)).thenReturn(user);
+
+        Optional<String> actualUserId = organisationService.findUserByEmail(AUTH_TOKEN_ID, USER_EMAIL);
+
+        assertThat(actualUserId).isEqualTo(Optional.of(user.getUserIdentifier()));
+    }
+
+    @Test
+    void shouldNotReturnUserIdIfUserNotPresent() {
+        Exception exception = new FeignException.NotFound(EMPTY, null);
+
+        when(organisationApi.findUsersByEmail(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID, USER_EMAIL)).thenThrow(exception);
+
+        Optional<String> actualUserId = organisationService.findUserByEmail(AUTH_TOKEN_ID, USER_EMAIL);
+
+        assertThat(actualUserId.isPresent()).isFalse();
+    }
+
+    @Test
+    void shouldRethrowExceptionOtherThanNotFound() {
+        Exception exception = new FeignException.InternalServerError(EMPTY, null);
+
+        when(organisationApi.findUsersByEmail(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID, USER_EMAIL)).thenThrow(exception);
+
+        try {
+            organisationService.findUserByEmail(AUTH_TOKEN_ID, USER_EMAIL);
```

Comment:
this can be wrapped in `assertThrows` rather than try-catch block

Reply:
done